pallets/runtime/tests/src/bridge.rs

```diff
@@ -12,790 +9,515 @@ use frame_support::{
     traits::{Currency, OnInitialize},
     weights::Weight,
 };
-use pallet_balances as balances;
-use pallet_bridge::{self as bridge, BridgeTx, BridgeTxDetail, BridgeTxStatus};
-use pallet_multisig as multisig;
-use polymesh_primitives::Signatory;
-use test_client::AccountKeyring;
+use pallet_bridge::{
+    self as bridge, BridgeTx as GBridgeTx, BridgeTxDetail as GBridgeTxDetail, BridgeTxStatus,
+};
+use sp_core::sr25519::Public;
+use test_client::AccountKeyring::*;
 
 type Bridge = bridge::Module<TestStorage>;
 type BridgeGenesis = bridge::GenesisConfig<TestStorage>;
 type Error = bridge::Error<TestStorage>;
-type Balances = balances::Module<TestStorage>;
-type MultiSig = multisig::Module<TestStorage>;
+type Identity = pallet_identity::Module<TestStorage>;
+type Balances = pallet_balances::Module<TestStorage>;
+type MultiSig = pallet_multisig::Module<TestStorage>;
 type Origin = <TestStorage as frame_system::Trait>::Origin;
 type System = frame_system::Module<TestStorage>;
 type Scheduler = pallet_scheduler::Module<TestStorage>;
 
-macro_rules! assert_tx_approvals_and_next_block {
-    ($address:expr, $proposal_id:expr, $num_approvals:expr) => {{
-        assert_eq!(
-            MultiSig::proposal_detail(&($address, $proposal_id)).approvals,
-            $num_approvals
-        );
-        next_block();
-    }};
-}
+type BridgeTx = GBridgeTx<Public, u128>;
+type BridgeTxDetail = GBridgeTxDetail<u128, u64>;
+
+const AMOUNT: u128 = 1_000_000_000;
+const AMOUNT_OVER_LIMIT: u128 = 1_000_000_000_000_000_000_000;
+const WEIGHT_EXPECTED: u64 = 500000210u64;
+const MIN_SIGNS_REQUIRED: u64 = 2;
+
+fn test_with_controller(test: &dyn Fn(&[Public])) {
```

Comment:
Not that it matters... this is in testing... but why the dynamic dispatch?

Reply:
As you said, this is just for functional testing, so I thought it would be better to use dynamic dispatching instead of static and do not increase build time (minimum but less static dispatch, less load to the compiler).
In production code, I would choice static dispatch (most likelihood)